xskillscore/core/comparative.py

```diff
@@ -167,3 +177,141 @@ def _categorical_metric(observations, forecasts, dim):
+        observations to be compared to both forecasts. if None, assumes that arguments
+        forecasts1 and forecasts2 are already MAEs. Defaults to None.
+    time_dim : str
+        time dimension of dimension over which to compute the temporal correlation.
+        Defaults to ``'time'``.
+    dim : str or list of str
+        dimensions to apply MAE to. Cannot contain ``time_dim``. Defaults to [].
+    alpha : float or 'return_p'
+        significance level alpha or return the significance level / p-value that
+        forecast1 is different than forecast2.
+
+    Returns
+    -------
+    xarray.DataArray or xarray.Dataset : Difference in xs.mae reduced by ``dim`` and
+        ``time_dim`` and:
+            * if ``alpha`` is float: half-width of the confidence interval at the
+             significance level ``alpha``.
+            * if ``alpha == 'return_p'``: the significance level that forecasts1 is
+             different than forecasts2.
+
+    Examples
+    --------
+    >>> f1 = xr.DataArray(np.random.normal(size=(30)),
+    ...      coords=[('time', np.arange(30))])
+    >>> f2 = xr.DataArray(np.random.normal(size=(30)),
+    ...      coords=[('time', np.arange(30))])
+    >>> o = xr.DataArray(np.random.normal(size=(30)),
+    ...      coords=[('time', np.arange(30))])
+    >>> mae_test(f1, f2, o, time_dim='time', dim=[], alpha=0.05)
+    <xarray.DataArray (results: 3)>
+    array([-0.09949737,  0.24421034,  0.        ])
+    Coordinates:
+        alpha    float64 0.05
+      * results  (results) <U11 'diff' 'hwci' 'significant'
+    >>> # absolute magnitude of difference is smaller than half-width of
+    >>> # confidence interval, therefore not significant at level alpha=0.05
+    >>> #
+    >>> mae_test(f1, f2, o, time_dim='time', dim=[], alpha='return_p')
+    <xarray.DataArray (results: 2)>
+    array([-0.09949737,  0.9226177])
+    Coordinates:
+      * results  (results) <U5 'diff' 'significance'
+    >>> # significance level of 0.92 much above commonly accepted level of 0.05
+
+
+    References
+    ----------
+        * https://www.cawcr.gov.au/projects/verification/CIdiff/FAQ-CIdiff.html
+
+    """
+    if isinstance(dim, str):
+        dim = [dim]
+    if time_dim in dim:
+        raise ValueError("`dim` cannot contain `time_dim`")
+
+    msg = f"`alpha` must be between 0 and 1 or `return_p`, found {alpha}"
+    if isinstance(alpha, str):
+        if alpha != "return_p":
+            raise ValueError(msg)
+    elif isinstance(alpha, float):
+        if not 0 < alpha < 1:
+            raise ValueError(msg)
+    else:
+        raise ValueError(msg)
+
+    if observations is not None:
+        # Compare the forecasts and observations using metric
+        mae_f1o = mae(observations, forecasts1, dim=dim)
+        mae_f2o = mae(observations, forecasts2, dim=dim)
+    elif observations is None:
+        mae_f1o = forecasts1
+        mae_f2o = forecasts2
+
+    pearson_r_f1f2 = pearson_r(mae_f1o, mae_f2o, dim=time_dim)
```

Comment:
Might be easy to just call this `rho = ...` since that's what the reference calls it.